tests/playwright/src/model/ai-lab-page.ts

```diff
@@ -0,0 +1,132 @@
+    this.recipesCatalogNaturalLanguageProcessing = this.recipesCatalogContent.getByRole('region', { name: 'Natural Language Processing', exact: true });
+    this.recipesCatalogAudio = this.recipesCatalogContent.getByRole('region', { name: 'Audio', exact: true });
+    this.recipesCatalogComputerVision = this.recipesCatalogContent.getByRole('region', { name: 'Computer Vision', exact: true });
+
+    this.applicationDetailsPanel = webview.getByLabel('application details panel');
+    this.startAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Start AI App' });
+    this.openAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Open AI App' });
+    this.deleteAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Delete AI App' });
+  }
+
+  async waitForPageLoad(): Promise<void> {
+    await playExpect(this.heading).toBeVisible();
+    await playExpect(this.navigationBar).toBeVisible();
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await playExpect(this.runningAppsButton).toBeVisible();
+    await playExpect(this.catalogButton).toBeVisible();
+    await playExpect(this.servicesButton).toBeVisible();
+    await playExpect(this.playgroundsButton).toBeVisible();
+  }
+
+  async openRecipesCatalog(): Promise<void> {
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await this.recipesCatalogButton.click();
+    await this.page.waitForTimeout(2000);
+    await playExpect(this.recipesCatalogPage).toBeVisible();
+    await playExpect(this.recipesCatalogContent).toBeVisible();
+    await playExpect(this.recipesCatalogNaturalLanguageProcessing).toBeVisible();
+    await playExpect(this.recipesCatalogAudio).toBeVisible();
+    await playExpect(this.recipesCatalogComputerVision).toBeVisible();
+  }
+
+  async openRecipesCatalogApp(category: Locator, appName: string): Promise<void> {
+    await this.openRecipesCatalog();
+    await category.getByRole('region', { name: 'content', exact: true }).first().getByRole('region', { name: appName, exact: true }).getByRole('button', { name: 'More details' }).click();
+    await playExpect(this.webview.getByRole('heading', { name: appName })).toBeVisible();
+  }
+
+  /**
+   * @throws {Error} If the app details are not visible or Delete AI App button is not visible
+   */
+  async tryDeleteOpenApp(): Promise<void> {
+    await playExpect(this.applicationDetailsPanel).toBeVisible();
+    console.log('Application details panel is visible, pressing delete button.');
+    await this.deleteAIAppButton.click();
+    try {
+      console.log('Waiting for Delete app confirmation dialog to appear.');
+      await playExpect(this.page.getByRole('dialog', { name: 'Podman AI Lab' })).toBeVisible();
+      console.log('Delete app confirmation dialog is visible, pressing Confirm button.');
+      await this.page.getByRole('dialog', { name: 'Podman AI Lab' }).getByRole('button', { name: 'Confirm' }).click();
```

Comment:
There is handleConfirmationDialog function in utilities in tests-playwright npm package.

Reply:
fixed.